grudge/array_context.py

```diff
@@ -0,0 +1,56 @@
+"""
+.. autoclass:: PyOpenCLArrayContext
+"""
+
+__copyright__ = "Copyright (C) 2020 Andreas Kloeckner"
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+
+from meshmode.array_context import PyOpenCLArrayContext as _PyOpenCLArrayContextBase
+from arraycontext.pytest import (
+        _PytestPyOpenCLArrayContextFactoryWithClass)
+
+
+class PyOpenCLArrayContext(_PyOpenCLArrayContextBase):
+    """Inherits from :class:`meshmode.array_context.PyOpenCLArrayContext`. Extends it
+    to understand :mod:`grudge`-specific transform metadata. (Of which there isn't
+    any, for now.)
+    """
+
+
+# {{{ pytest actx factory
+
+class PytestPyOpenCLArrayContextFactory(
+        _PytestPyOpenCLArrayContextFactoryWithClass):
+    actx_class = PyOpenCLArrayContext
+
+
+# deprecated
+class PytestPyOpenCLArrayContextFactoryWithHostScalars(
+        _PytestPyOpenCLArrayContextFactoryWithClass):
+    actx_class = PyOpenCLArrayContext
+    force_device_scalars = False
+
```

Comment:
```suggestion

register_pytest_array_context_factory("grudge.pyopencl",
        PytestPyOpenCLArrayContextFactory)
```

Reply:
Done as part of this force-push: https://github.com/inducer/grudge/compare/b8eaa66bc10f8e659f902ad11da4962b7bbef0ab..335d27cfc84f222795693d8d6d90bfab21be0247